src/engine/thread.js

```diff
@@ -254,17 +270,20 @@ class Thread {
      * @param {string} blockId Block ID to push to stack.
      */
     reuseStackForNextBlock (blockId) {
-        this.stack[this.stack.length - 1] = blockId;
-        this.stackFrames[this.stackFrames.length - 1].reuse();
+        this.pointer = blockId;
+        if (this.stackFrame.needReset) this.stackFrame.reuse();
```

Comment:
It looks like this is the only place in the vm that `reuse` is being used. Should we just consolidate `reuse` and `reset` now given that they have the same functionality?

Reply:
I think so yes.